ArgusProbe_Beamlogic.py

```diff
@@ -227,6 +228,284 @@ def _get_ntp_timestamp():
+            try:
+                with self.dataLock:
+                    pleaseConnect = self.pleaseConnect
+
+                if pleaseConnect:
+                    # open serial port
+                    self.serialHandler = serial.Serial(self.serialport, baudrate=self.baudrate)
+
+                    # read byte
+                    while True:
+                        waitingbytes   = self.serialHandler.inWaiting()
+                        if waitingbytes != 0:
+                            c= self.serialHandler.read(waitingbytes)
+                            for byte in c:
+                               self._newByte(byte)
+                            time.sleep(2)
+
+            except serial.SerialException:
+                # mote disconnected, or pyserialHandler closed
+                # destroy pyserial instance
+                print "WARNING: Could not read from serial at \"{0}\".".format(
+                       self.serialport)
+                print "Is device connected?"
+                self.goOn            = False
+                self.serialHandler   = None
+                time.sleep(1)
+
+            except Exception as err:
+                logCrash(self.name, err)
+
+    #======================== public ==========================================
+
+    def connectSerialPort(self):
+        with self.dataLock:
+            self.pleaseConnect = True
+
+    def disconnectSerialPort(self):
+        with self.dataLock:
+            self.pleaseConnect = False
+        try:
+            self.serialHandler.close()
+        except:
+            pass
+
+    def close(self):
+        self.goOn            = False
+    #======================== public ==========================================
+
+    #======================== private =========================================
+    def _rx_buf_add(self, byte):
+        """ Adds byte to buffer and escapes the XONXOFF bytes """
+
+        if byte == chr(self.XONXOFF_ESCAPE):
+            self.xonxoff_escaping = True
+        else:
+            if self.xonxoff_escaping is True:
+                self.rxBuffer.append(chr(ord(byte) ^ self.XONXOFF_MASK))
+                self.xonxoff_escaping = False
+            elif byte != chr(self.XON) and byte != chr(self.XOFF):
+                self.rxBuffer.append(byte)
+
+    def _handle_frame(self):
+        """ Handles a HDLC frame """
+        valid_frame = False
+        try:
+            self.rxBuffer  = self.hdlc.dehdlcify(self.rxBuffer)
+
+            if self.send_to_parser:
+                self.send_to_parser([ord(c) for c in self.rxBuffer])
+
+            if self.rxBuffer[0] == 'P':   #packet from sniffer SERFRAME_MOTE2PC_SNIFFED_PACKET 'P'
+                self.rxBuffer = self.rxBuffer[1:]
+                valid_frame   = True
+
+        except openhdlc.HdlcException as err:
+            #log.warning('{}: invalid serial frame: {} {}'.format(self.name, format_string_buf(temp_buf), err))
+            print 'HDLC Exception'
```

Comment:
this is too generic of a print, add some context like in the commented warning log just above